### DISCUSSION TITLE
Some useful warnings are missing. How to get them?

### DISCUSSION THREAD

**Discussion by ylazy** [BowlerHatLLC/vscode-as3mxml#758] [Q&A] [answered]
Hi!

When I compile my app with Adobe Animate, I got many useful warnings:

`Warning: 1100: Assignment within conditional.  Did you mean == instead of =?`
![image](https://github.com/user-attachments/assets/c5ca4c27-676f-455d-8802-85ab6a8d79ef)

`Warning: 3592: startAt is not a recognized property of the dynamic class RegExp.`
![image](https://github.com/user-attachments/assets/78b1c043-2f0e-42df-a403-48dbbab5ea39)

`Warning: 3590: int used where a Boolean value was expected.  The expression will be type coerced to Boolean.`
![image](https://github.com/user-attachments/assets/c20f6619-8857-4871-82c2-c41e37e895ca)

`Warning: 1102: null used where a Number value was expected.`
![image](https://github.com/user-attachments/assets/d729084a-37d7-406d-9fb3-4de0439492f7)

`Warning: 3590: Array used where a Boolean value was expected.  The expression will be type coerced to Boolean.`
![image](https://github.com/user-attachments/assets/328ed249-93f6-4201-8adb-efe49db127bd)

...

But I can't see them if I compile using the Royale Compiler.
If I missed something?

Thank you!

**Comment by joshtynjala** [comment, marked as answer]
> Warning: 1100: Assignment within conditional.  Did you mean == instead of =?

I can see this warning in vscode-as3mxml if it's in an `if`, but not in an `else if`. Definitely a royale-compiler bug.

> Warning: 3592: startAt is not a recognized property of the dynamic class RegExp.

I confirm that I don't see a warning for this one.

This one is interesting. A dynamic class is intended to support additional properties that it does not define. It seems strange to me that setting a property like this would result in a warning. The ASC 2.0 compiler does not warn for it either, but the Flex SDK compiler does.

 I would lean toward not adding this warning to royale-compiler.

> Warning: 3590: int used where a Boolean value was expected.  The expression will be type coerced to Boolean.
> Warning: 3590: Array used where a Boolean value was expected.  The expression will be type coerced to Boolean.

I confirm that I don't see a warning for this one. The ASC 2.0 compiler does not warn for it either, but the Flex SDK compiler does.

This seems like a potentially useful warning to add to royale-compiler.

> Warning: 1102: null used where a Number value was expected.

I confirm that I don't see a warning for this one. The ASC 2.0 compiler does not warn for it either, but the Flex SDK compiler does.

This seems like a potentially useful warning to add to royale-compiler.
